### ISSUE TITLE
[False-Positive] Variable "structure" not used

### ISSUE THREAD

**Issue by lucasborin** [abaplint/abaplint#1889]
![image](https://user-images.githubusercontent.com/5233413/113759234-3b38ef00-96eb-11eb-8ef9-2be086e9383e.png)

![image](https://user-images.githubusercontent.com/5233413/113759279-468c1a80-96eb-11eb-9d17-a1d9b88b3602.png)

![image](https://user-images.githubusercontent.com/5233413/113759319-51df4600-96eb-11eb-821e-1fefdd494d3c.png)

Run: https://github.com/SAP/code-pal-for-abap/pull/304/checks?check_run_id=2280892645

**Comment by larshp** [comment]
```abap
  METHOD get_back_statement.
    TRY.
        DATA(back_structure) = ref_scan_manager->structures[ structure-back ].
        result = ref_scan_manager->statements[ back_structure-stmnt_from ].
      CATCH cx_sy_itab_line_not_found.
        CLEAR result.
    ENDTRY.
  ENDMETHOD.
```

**Comment by larshp** [comment]
reproduced as

```abap
DATA ref_scan_manager TYPE REF TO sdfsdfsd.
DATA(back_structure) = ref_scan_manager->structures[ structure-back ].
DATA(sdfs) = ref_scan_manager->statements[ back_structure-stmnt_from ].
WRITE sdfs.
```

![image](https://user-images.githubusercontent.com/5888506/113815033-d88b3600-9772-11eb-8db5-c52a58bb29bf.png)

**Comment by larshp** [comment]
@lucasborin try again, 2.73.6 published

**Comment by lucasborin** [comment]
One more similar case here:

![image](https://user-images.githubusercontent.com/5233413/113903446-6254f680-97a7-11eb-9367-4a781b0410f5.png)

![image](https://user-images.githubusercontent.com/5233413/113903479-6aad3180-97a7-11eb-91f6-94d07f9ba1e8.png)

Run: https://github.com/SAP/code-pal-for-abap/pull/304/checks?check_run_id=2289434137

**Comment by lucasborin** [comment]
![image](https://user-images.githubusercontent.com/5233413/113904559-8e24ac00-97a8-11eb-9798-e0fb5e3e0198.png)

![image](https://user-images.githubusercontent.com/5233413/113904537-87963480-97a8-11eb-9405-5f67ba64e0b1.png)

Abstract methods are not relevant.

Run: https://github.com/SAP/code-pal-for-abap/pull/304/checks?check_run_id=2289434137

**Comment by larshp** [comment]
previous reproduced as 

```abap
DATA result TYPE STANDARD TABLE OF string.
DATA int TYPE i VALUE 1.
result[ int ] = 'hello'.
```

**Comment by lucasborin** [comment]
One more case:

![image](https://user-images.githubusercontent.com/5233413/113911390-4e61c280-97b0-11eb-85ee-fca4cdeb5b5e.png)

![image](https://user-images.githubusercontent.com/5233413/113911432-591c5780-97b0-11eb-9ab5-4d3c2212eb02.png)

Run: https://github.com/SAP/code-pal-for-abap/pull/304/checks?check_run_id=2289709342

**Comment by larshp** [comment]
suggest moving 
https://github.com/abaplint/abaplint/issues/1889#issuecomment-815068903
https://github.com/abaplint/abaplint/issues/1889#issuecomment-815104902
to 2 new separate issues, I'm having trouble keeping track

rest should be fixed in the latest release

**Comment by larshp** [comment]
cool, then I guess this issue can be closed?
